### PR TITLE
Add more solutions

### PR DESCRIPTION
**Pull request by sidaksohi**

### CODE REVIEW

**Review by vbrazo — CHANGES_REQUESTED** [review]
Could we split this PR into multiple PRs and also add an appropriate description at the top of each algorithm?

**Comment by sidaksohi** [comment]
Sure.

File: maths/two-sum.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# @return {Integer[]}
+def two_sum(nums, target)
+    nums_hash = Hash.new
+    
+    nums.each_with_index do |val, ind|
+      complement = target - val
+      return [ind, nums_hash[complement]] if nums_hash[complement]
+      nums_hash[val] = ind
+    end
+end
```
Comment (vbrazo) [inline maths/two-sum.rb:12]: We have this algorithm:
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/arrays/two_sum.rb
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/arrays/two_sum_ii.rb
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/hash_table/two_sum.rb

Comment (sidaksohi) [reply maths/two-sum.rb:12]: Good catch, thank you. I'll start now.

Comment (vbrazo) [reply maths/two-sum.rb:12]: Thanks. I'll close this PR for now.

**Pull request closed without merging**